target/linux/ath79/dts/ar7161_mikrotik_routerboard-493ah.dts

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: GPL-1.0-or-later OR MIT
+/dts-v1/;
+
+#include "ar7100_mikrotik_routerboard-4xx.dtsi"
+
+/ {
+	compatible = "mikrotik,routerboard-493", "qca,ar7161";
+	model = "MikroTik RouterBOARD 493AH";
+
+	aliases {
+		mdio-gpio0 = &mdio_gpio0;
+	};
+};
+
+&eth1 {
```

Comment:
I think you can delete this, and the alias above, as this only has 1 Ethernet switch.

Reply:
Also, I think you can add this after removing `&eth1` as gpio7 is available.

```
keys: keys {                                
    compatible = "gpio-keys";               
                                            
    status = "disabled";                    
                                            
    reset_key: reset {                      
        label = "reset";                    
        linux,code = <KEY_RESTART>;         
        gpios = <&gpio 7 GPIO_ACTIVE_LOW>;  
        debounce-interval = <60>;           
    };                                      
};                                          
```

Reply:
I was able to netboot an ath79 kernel, and this is big progress. The router boots, but neither Ethernet interfaces seem to work. I tried both with my original configuration and after removing the definition of eth1 as @ch6574 suggested. The switch also seems undetected; running `swconfig list` does not list anything.

I saw this in `dmesg` when booting my ath79 kernel:
```
[    6.367446] ag71xx 19000000.eth: invalid MAC address, using random address
[    6.713850] ag71xx 19000000.eth: connected to PHY at mdio.0:00 [uid=02430d80, driver=Gener]
[    6.723463] eth0: Atheros AG71xx at 0xb9000000, irq 4, mode: rgmii
[    6.729954] ag71xx 1a000000.eth: invalid MAC address, using random address
[    7.063880] ag71xx 1a000000.eth: Could not connect to PHY device. Deferring probe.
[    7.124519] ag71xx 1a000000.eth: invalid MAC address, using random address
[    7.448389] ag71xx 1a000000.eth: Could not connect to PHY device. Deferring probe.
[    7.456397] ag71xx 1a000000.eth: invalid MAC address, using random address
[    7.780378] ag71xx 1a000000.eth: Could not connect to PHY device. Deferring probe.
[    8.328963] IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
[   15.564619] ag71xx 1a000000.eth: invalid MAC address, using random address
[   15.888489] ag71xx 1a000000.eth: Could not connect to PHY device. Deferring probe.
[   30.680785] br-lan: port 1(eth0) entered blocking state
[   30.686079] br-lan: port 1(eth0) entered disabled state
[   30.691659] device eth0 entered promiscuous mode
[  268.458696] device eth0 left promiscuous mode
[  268.463230] br-lan: port 1(eth0) entered disabled state
[  268.498295] IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
```

To clarify, I think the RB493AH has two Ethernet interfaces but one switch interface. To compare, booting an old ar71xx kernel says:
```
[    6.930000] eth0: Atheros AG71xx at 0xb9000000, irq 4                        
[    7.240000] eth1: Atheros AG71xx at 0xba000000, irq 5
```

Reply:
The above dmesg shows eth0 working (it's 19000000.eth), and eth1 not (hence my comment to delete it).

Do you not see eth0 as an interface?

N.B. You might find the VLAN settings need adjusting.

Reply:
As for your eth1, you'll need to set it up as per https://github.com/openwrt/openwrt/blob/835c932ccb6bb43ea94fb28a76fca2019a2f917c/target/linux/ar71xx/files/arch/mips/ath79/mach-rb4xx.c#L389.

Notice it's not the same mdio gpio setup the 493g has.

Reply:
I forgot to mention, you might need this for your eth0

```
&eth0 {
	phy-mode = "mii";
};
```

and perhaps something as simple as this will work for eth1 (I have no way to test it out, so this is guesswork)

```
&mdio0 {
	phy1: ethernet-phy@1{
		reg = <1>;
	};
};

&eth1 {
	status = "okay";
	
	phy-mode = "rmii";
	phy-handle = <&phy1>;
};
```